Source/MoreDotNet.Test/Extensions/Common/RandomExtensions/OneOfTests.cs

```diff
@@ -0,0 +1,33 @@
+﻿namespace MoreDotNet.Tests.Extensions.Common.RandomExtensions
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+
+    using MoreDotNet.Extensions.Common;
+
+    using Xunit;
+
+    public class OneOfTests
+    {
+        [Fact]
+        public void OneOf_ShouldReturnOneOfIntArray()
+        {
+            var random = new Random();
+            var numbers = new[] { 1, 2, 3, 4, 5, 6 };
+            var number = random.OneOf<int>(numbers);
+
+            Assert.True(numbers.Contains(number));
+        }
+
+        [Fact]
+        public void OneOf_ShouldReturnOneOfIntCollection()
```

Comment:
Testing for different types of collections is unnecessary. We are testing our method not the implementations of Arrays, List, etc.